ui/desktop/src/components/ui/CustomRadio.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+
+/**
+ * CustomRadio - A reusable radio button component with dark mode support
+ * @param {Object} props - Component props
+ * @param {string} props.id - Unique identifier for the radio input
+ * @param {string} props.name - Name attribute for the radio input
+ * @param {string} props.value - Value of the radio input
+ * @param {boolean} props.checked - Whether the radio is checked
+ * @param {function} props.onChange - Function to call when radio selection changes
+ * @param {boolean} [props.disabled] - Whether the radio is disabled
+ * @param {React.ReactNode} [props.label] - Primary label content
+ * @param {React.ReactNode} [props.secondaryLabel] - Secondary/subtitle label content
+ * @param {React.ReactNode} [props.rightContent] - Optional content to display on the right side
+ * @param {string} [props.className] - Additional CSS classes for the main container
+ * @returns {JSX.Element}
+ */
+const CustomRadio = ({
```

Comment:
We should consider if we should use this, and if we can unify with what I had here for the top level of the settings page model radio buttons https://github.com/block/goose/blob/2f00765abd4459dc7da43b27e1978dfc83c65889/ui/desktop/src/components/settings_v2/SettingsView.tsx#L90-L96